src/library/scala/collection/mutable/PriorityQueue.scala

```diff
@@ -254,12 +320,24 @@ sealed class PriorityQueue[A](implicit val ord: Ordering[A])
    *
    *  @return  a priority queue with the same elements.
    */
-  override def clone(): PriorityQueue[A] = new PriorityQueue[A] ++= this.iterator
+  override def clone(): PriorityQueue[A] = {
+    val pq = new PriorityQueue[A]
+    for (i <- 1 to this.length) pq.unsafeAdd(resarr(i))
+    pq
+  }
 }
 
 
 object PriorityQueue extends OrderedTraversableFactory[PriorityQueue] {
-  def newBuilder[A](implicit ord: Ordering[A]) = new PriorityQueue[A]
+  def newBuilder[A](implicit ord: Ordering[A]): Builder[A, PriorityQueue[A]] = {
+    new Builder[A, PriorityQueue[A]] {
+      val pq = new PriorityQueue[A]
+      def +=(elem: A): this.type = { pq.unsafeAdd(elem); this }
+      def result(): PriorityQueue[A] = { pq.heapify(1); pq }
```

Comment:
I think you're suggesting a behavior where you can add 1,2,3 to the builder, call result to get a priority queue with elements 1,2,3, then add 4,5,6 to the builder, and call result again to get a priority queue with elements 1,2,3,4,5,6.  But the priority queues are mutable, and I don't think you would want adding 4 to the builder to retroactively also add 4 to the first priority queue.  Or maybe I'm misunderstanding what you are saying.